openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/insights/DataInsightsApp.java

```diff
@@ -1,17 +1,112 @@
+import org.openmetadata.schema.entity.services.ingestionPipelines.PipelineType;
+import org.openmetadata.schema.metadataIngestion.DataInsightPipeline;
+import org.openmetadata.schema.metadataIngestion.SourceConfig;
+import org.openmetadata.schema.type.EntityReference;
+import org.openmetadata.schema.type.ProviderType;
+import org.openmetadata.schema.type.Relationship;
+import org.openmetadata.service.Entity;
 import org.openmetadata.service.apps.AbstractNativeApplication;
+import org.openmetadata.service.exception.EntityNotFoundException;
 import org.openmetadata.service.jdbi3.CollectionDAO;
+import org.openmetadata.service.jdbi3.EntityRepository;
+import org.openmetadata.service.jdbi3.IngestionPipelineRepository;
 import org.openmetadata.service.search.SearchRepository;
+import org.openmetadata.service.util.FullyQualifiedName;
 
 @Slf4j
 public class DataInsightsApp extends AbstractNativeApplication {
 
+  private static final String INGESTION_PIPELINE_NAME = "OpenMetadata_dataInsight";
+  private static final String SERVICE_NAME = "OpenMetadata";
+  private static final String SERVICE_TYPE = "Metadata";
+  private static final String PIPELINE_DESCRIPTION = "OpenMetadata DataInsight Pipeline";
+
   @Override
   public void init(App app, CollectionDAO dao, SearchRepository searchRepository) {
     super.init(app, dao, searchRepository);
     LOG.info("Data Insights App is initialized");
   }
+
+  @Override
+  public void initializeExternalApp() {
+    if (getApp().getAppType() == AppType.External && getApp().getScheduleType().equals(ScheduleType.Scheduled)) {
+      IngestionPipelineRepository ingestionPipelineRepository =
+          (IngestionPipelineRepository) Entity.getEntityRepository(Entity.INGESTION_PIPELINE);
+
+      try {
+        // Check if the Pipeline Already Exists
+        String fqn = FullyQualifiedName.add(SERVICE_NAME, INGESTION_PIPELINE_NAME);
+        IngestionPipeline storedPipeline =
+            ingestionPipelineRepository.getByName(null, fqn, ingestionPipelineRepository.getFields("id"));
+
+        // Init Application Code for Some Initialization
+        List<CollectionDAO.EntityRelationshipRecord> records =
+            collectionDAO
+                .relationshipDAO()
+                .findTo(getApp().getId(), Entity.APPLICATION, Relationship.HAS.ordinal(), Entity.INGESTION_PIPELINE);
+
+        if (records.isEmpty()) {
+          // Add Ingestion Pipeline to Application
+          collectionDAO
+              .relationshipDAO()
+              .insert(
+                  getApp().getId(),
+                  storedPipeline.getId(),
+                  Entity.APPLICATION,
+                  Entity.INGESTION_PIPELINE,
+                  Relationship.HAS.ordinal());
+        }
+      } catch (EntityNotFoundException ex) {
+        // Pipeline needs to be created
+        EntityRepository<?> serviceRepository = Entity.getServiceEntityRepository(ServiceType.fromValue(SERVICE_TYPE));
```

Comment:
lets move these into meaningful methods, instead of having one big method lot of lines